File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int n = s.size();
        int i = 0;
        string result = "";
        while(i<n){
            while(i<n && s[i]==' '){
                i++;
            }
            if(i>=n){
                break;
            }
            int j = i+1;
            while(j<n && s[j]!=' '){
                j++;
            }
            string str = s.substr(i, j-i);
            if(result.size()==0){
                result = str;
            }
            else{
                result = str + ' ' + result;
            }
            i = j + 1;
        }
        return result;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        int n = s.length();
        int i = 0;
        StringBuilder result = new StringBuilder();
        while(i<n){
            while(i<n && s.charAt(i)==' '){
                i++;
            }
            if(i>=n){
                break;
            }
            int j = i + 1;
            while(j<n && s.charAt(j)!=' '){
                j++;
            }
            String str = s.substring(i, j);
            if(result.length()==0){
                result.append(str);
            }
            else{
                result.insert(0, str+' ');
            }
            i = j + 1;
        }
        return result.toString();
    }
}